be/src/storage/rowset/column_iterator_wrapper.h

```diff
@@ -0,0 +1,110 @@
+
+    Status next_batch(size_t* n, Column* dst) override { return _iter->next_batch(n, dst); }
+
+    Status next_batch(const SparseRange<>& range, Column* dst) override { return _iter->next_batch(range, dst); }
+
+    ordinal_t get_current_ordinal() const override { return _iter->get_current_ordinal(); }
+
+    Status get_row_ranges_by_zone_map(const std::vector<const ColumnPredicate*>& predicates,
+                                      const ColumnPredicate* del_predicate, SparseRange<>* row_ranges) override {
+        return _iter->get_row_ranges_by_zone_map(predicates, del_predicate, row_ranges);
+    }
+
+    Status get_row_ranges_by_bloom_filter(const std::vector<const ColumnPredicate*>& predicates,
+                                          SparseRange<>* row_ranges) override {
+        return _iter->get_row_ranges_by_bloom_filter(predicates, row_ranges);
+    }
+
+    bool all_page_dict_encoded() const override { return _iter->all_page_dict_encoded(); }
+
+    Status fetch_all_dict_words(std::vector<Slice>* words) const override { return _iter->fetch_all_dict_words(words); }
+
+    int dict_size() override { return _iter->dict_size(); }
+
+    int dict_lookup(const Slice& word) override { return _iter->dict_lookup(word); }
+
+    Status next_dict_codes(size_t* n, Column* dst) override { return _iter->next_dict_codes(n, dst); }
+
+    Status next_dict_codes(const SparseRange<>& range, Column* dst) override {
+        return _iter->next_dict_codes(range, dst);
+    }
+
+    Status decode_dict_codes(const int32_t* codes, size_t size, Column* words) override {
+        return _iter->decode_dict_codes(codes, size, words);
+    }
+
+    int64_t element_ordinal() const override { return _iter->element_ordinal(); }
+
+    Status seek_to_ordinal_and_calc_element_ordinal(ordinal_t ord) override {
+        return _iter->seek_to_ordinal_and_calc_element_ordinal(ord);
+    }
+
+    Status decode_dict_codes(const Column& codes, Column* words) override {
+        return _iter->decode_dict_codes(codes, words);
+    }
+
+    Status fetch_values_by_rowid(const rowid_t* rowids, size_t size, Column* values) override {
+        return _iter->fetch_values_by_rowid(rowids, size, values);
+    }
+
+    Status fetch_dict_codes_by_rowid(const rowid_t* rowids, size_t size, Column* values) override {
+        return _iter->fetch_dict_codes_by_rowid(rowids, size, values);
+    }
+
+    Status next_batch(size_t* n, Column* dst, ColumnAccessPath* path) override {
+        return _iter->next_batch(n, dst, path);
+    }
+
+    Status next_batch(const SparseRange<>& range, Column* dst, ColumnAccessPath* path) override {
+        return _iter->next_batch(range, dst, path);
+    }
+
+    Status fetch_subfield_by_rowid(const rowid_t* rowids, size_t size, Column* values) override {
+        return _iter->fetch_subfield_by_rowid(rowids, size, values);
+    }
+
+protected:
+    ColumnIterator* _iter;
+    Ownership _ownership;
```

Comment:
`_ownership` can be eliminated, use `unique_ptr` to explicitly manages the lifecycle and ownership.

Reply:
The ownership is non-deterministic.

Reply:
I mean the unique_ptr can be initialized based on the ownship passed in constructor, so in destructor, you don't have to do anything, everything is settled when the contruction done.

Reply:
Updated